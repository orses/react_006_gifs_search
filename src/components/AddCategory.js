import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = e => {
    setInputValue(e.target.value);
  };

  const handleSubmit = e => {
    e.preventDefault();
    // updating categories state
    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="search"
        value={inputValue}
        onChange={handleInputChange}
        placeholder="Type a category and hit Enter"
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
