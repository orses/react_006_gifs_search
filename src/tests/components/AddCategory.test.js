import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('test on AddCategory', () => {
  const setCategories = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change the value on the input field', () => {
    const input = wrapper.find('input');
    input.simulate('change', {
      target: {
        value: 'Changing data',
      },
    });
  });

  test('should not submit data if input value are empty', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call setCategories and clean up the input field', () => {
    // 1ยบ simulate input change
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: 'New data in input field' } });

    // 2ยบ simulate a submit action
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    // 3ยบ setCategories must being called one time and we a function
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    // 4ยบ set input field as empty
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
