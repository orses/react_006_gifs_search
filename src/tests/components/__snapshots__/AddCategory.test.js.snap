// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on AddCategory should render the component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Type a category and hit Enter"
    type="search"
    value=""
  />
</form>
`;
